src/python/pants/backend/jvm/tasks/jvm_platform_analysis.py

```diff
@@ -181,8 +185,45 @@ def validate_platform_dependencies(self):
     """
     conflicts = []
 
+    jvm_platform_map = {}
+
+    class JvmPlatformAssignment(datatype([
+        ('from_upstream', bool),
+        ('source', SubclassesOf(JvmTarget)),
+        'version',
```

Comment:
What datatype is this? String like `1.8` or `Revision`? I recommend adding a type constraint.

Reply:
Done!